src/main/java/org/mifos/connector/ams/paygops/zeebe/ZeebeWorkers.java

```diff
@@ -68,15 +69,24 @@ public void setupWorkers() {
                         producerTemplate.send("direct:transfer-validation-base", ex);
 
                         boolean isPartyLookUpFailed = ex.getProperty(PARTY_LOOKUP_FAILED, boolean.class);
-                        variables.put(PARTY_LOOKUP_FAILED, isPartyLookUpFailed);
+                        if(isPartyLookUpFailed) {
+                            variables.put(PARTY_LOOKUP_FAILED, true);
```

Comment:
Line 73 to 76 is more or less duplicated everywhere. This should be separated as a function reused in all the 4 places.

Error information manually populated can be handled through control statements inside that function.